File: ga/gpt3_5_result_2024_6/result/1968B_cpp_gpt/3_1968B.cpp
use std::io::{self, BufRead};

fn main() {
	let stdin = io::stdin();
	let mut input = stdin.lock();
	
	let mut t = String::new();
	input.read_line(&mut t).unwrap();
	let t: i32 = t.trim().parse().unwrap();
	
	for _ in 0..t {
		let mut line = String::new();
		input.read_line(&mut line).unwrap();
		let mut values = line.split_whitespace().map(|x| x.parse::<i32>().unwrap());
		let x = values.next().unwrap();
		let k = values.next().unwrap();
		
		let mut a = String::new();
		input.read_line(&mut a).unwrap();
		let a = a.trim();
		
		let mut b = String::new();
		input.read_line(&mut b).unwrap();
		let b = b.trim();
		
		let mut ans = 0;
		for i in b.chars() {
			if let Some(c) = a.chars().nth(ans as usize) {
				if c == i {
					ans += 1;
				}
			}
		}
		
		println!("{}", ans);
	}
}

File: ga/glm4_result_2024_6/result/1968B_cpp_glm/3_1968B.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t = input.trim().parse::<i32>().unwrap();

    for _ in 0..t {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let mut iter = input.trim().split_whitespace();
        let x = iter.next().unwrap().parse::<i32>().unwrap();
        let k = iter.next().unwrap().parse::<i32>().unwrap();
        
        let mut a = String::new();
        io::stdin().read_line(&mut a).unwrap();
        let mut b = String::new();
        io::stdin().read_line(&mut b).unwrap();

        let mut ans = 0;
        for i in b.chars() {
            if ans < a.len() && a.chars().nth(ans).unwrap() == i {
                ans += 1;
            }
        }

        println!("{}", ans);
    }
}